core/.js/src/test/scala/io/udash/testing/TestViewFactory.scala

```diff
@@ -14,31 +15,42 @@ class TestViewFactory[T <: TestState] extends ViewFactory[T] {
   }
 }
 
-class TestView extends ContainerView {
+class TestView(overrideContent: OptArg[String] = OptArg.Empty) extends ContainerView {
+
   import scalatags.JsDom.all._
+
   var lastChild: View = _
   var renderingCounter = 0
   var closed = false
 
+  private def content: String = overrideContent.getOrElse(super.toString)
+
   override def renderChild(view: Option[View]): Unit = {
     super.renderChild(view)
     lastChild = view.orNull
   }
 
+  override def clearChildViewContainer(): Unit = {
+    super.clearChildViewContainer()
+    lastChild = null
```

Comment:
Is that necessary? The only way to call `ContainerView#clearChildViewContainer` (except for subclasses) goes through a `ContainerView#renderChild` invocation, which handles `lastChild` properly.

In fact, I'd consider making `ContainerView#clearChildViewContainer` private.

Reply:
`clearChildViewContainer` cannot be private - it is a hook which may be used by `ContainerView` implementations in the wild and the only clean way to attach user code to container cleanup.

This change is not necessary here, but it's factually correct. In fact `renderChild` does too much if it's supposed to resemble real interaction - it should only do `view.foreach`. I found this out while testing different approaches to handling this issue and I think lying in these tests makes it harder to debug these cases.